### ISSUE TITLE
expo-cli: TypeError: Cannot read property 'compile' of undefined

### ISSUE THREAD

**Issue by TheSalarKhan** [expo/expo-cli#213]
## Description

Expo-cli is crashing on npm start with the error message 
```[02:46:59] TypeError: Cannot read property 'compile' of undefined```

## Expected Behavior

1. install expo-cli using `npm install expo-cli -g`
2. start a new project using 'expo init`
3. Browser pops up with dev tools and a QR code is visible on the terminal.

## Observed Behavior

1. The browser opens up blank
2. The terminal reads:
```
msalar@Dranzer:~$ expo init myproj
? Choose a template: blank
[02:57:27] Extracting project files...
[02:57:28] Customizing project...

Your project is ready at /home/msalar/myproj
To get started, you can type:

  cd myproj
  expo start

msalar@Dranzer:~$ cd myproj/
msalar@Dranzer:~/myproj$ npm start

> empty-project-template@ start /home/msalar/myproj
> expo start

[03:01:33] Starting project at /home/msalar/myproj
[03:01:33] Expo DevTools is running at http://localhost:19002
[03:01:33] Opening DevTools in the browser... (press shift-d to disable)
[03:01:37] TypeError: Cannot read property 'compile' of undefined
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! empty-project-template@ start: `expo start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the empty-project-template@ start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/msalar/.npm/_logs/2018-11-25T22_01_37_331Z-debug.log
msalar@Dranzer:~/myproj$ 
```

Here's the log file
```
0 info it worked if it ends with ok
1 verbose cli [ '/home/msalar/.nvm/versions/node/v8.12.0/bin/node',
1 verbose cli   '/home/msalar/.nvm/versions/node/v8.12.0/bin/npm',
1 verbose cli   'start' ]
2 info using npm@6.4.1
3 info using node@v8.12.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle empty-project-template@~prestart: empty-project-template@
6 info lifecycle empty-project-template@~start: empty-project-template@
7 verbose lifecycle empty-project-template@~start: unsafe-perm in lifecycle true
8 verbose lifecycle empty-project-template@~start: PATH: /home/msalar/.nvm/versions/node/v8.12.0/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/msalar/myproj/node_modules/.bin:/home/msalar/.nvm/versions/node/v8.12.0/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
9 verbose lifecycle empty-project-template@~start: CWD: /home/msalar/myproj
10 silly lifecycle empty-project-template@~start: Args: [ '-c', 'expo start' ]
11 silly lifecycle empty-project-template@~start: Returned: code: 1  signal: null
12 info lifecycle empty-project-template@~start: Failed to exec start script
13 verbose stack Error: empty-project-template@ start: `expo start`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/home/msalar/.nvm/versions/node/v8.12.0/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
13 verbose stack     at emitTwo (events.js:126:13)
13 verbose stack     at EventEmitter.emit (events.js:214:7)
13 verbose stack     at ChildProcess.<anonymous> (/home/msalar/.nvm/versions/node/v8.12.0/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at emitTwo (events.js:126:13)
13 verbose stack     at ChildProcess.emit (events.js:214:7)
13 verbose stack     at maybeClose (internal/child_process.js:915:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
14 verbose pkgid empty-project-template@
15 verbose cwd /home/msalar/myproj
16 verbose Linux 4.15.0-29-generic
17 verbose argv "/home/msalar/.nvm/versions/node/v8.12.0/bin/node" "/home/msalar/.nvm/versions/node/v8.12.0/bin/npm" "start"
18 verbose node v8.12.0
19 verbose npm  v6.4.1
20 error code ELIFECYCLE
21 error errno 1
22 error empty-project-template@ start: `expo start`
22 error Exit status 1
23 error Failed at the empty-project-template@ start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

## Environment
It was giving errors so I set EXPO_DEBUG=true.
```
msalar@Dranzer:~/myproj$ expo diagnostics
[03:03:48] Generating diagnostics report...
[03:03:48] You can join our slack here: https://slack.expo.io/.

  Expo CLI 2.4.0 environment info:
    System:
      OS: Linux 4.15 Ubuntu 18.04.1 LTS (Bionic Beaver)
      Shell: 4.4.19 - /bin/bash
    Binaries:
      Node: 8.12.0 - ~/.nvm/versions/node/v8.12.0/bin/node
      npm: 6.4.1 - ~/.nvm/versions/node/v8.12.0/bin/npm
    npmPackages:
      expo: ^31.0.2 => 31.0.2 
      react: 16.5.0 => 16.5.0 
      react-native: https://github.com/expo/react-native/archive/sdk-31.0.0.tar.gz => 0.57.1 
    npmGlobalPackages:
      expo-cli: 2.4.0

  Diagnostics report:
/[03:03:50] Request failed with status code 500
[03:03:50] Error: Request failed with status code 500
    at createError (/home/msalar/.nvm/versions/node/v8.12.0/lib/node_modules/expo-cli/node_modules/axios/lib/core/createError.js:16:15)
    at settle (/home/msalar/.nvm/versions/node/v8.12.0/lib/node_modules/expo-cli/node_modules/axios/lib/core/settle.js:17:12)
    at IncomingMessage.handleStreamEnd (/home/msalar/.nvm/versions/node/v8.12.0/lib/node_modules/expo-cli/node_modules/axios/lib/adapters/http.js:236:11)
    at emitNone (events.js:111:20)
    at IncomingMessage.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1064:12)
    at _combinedTickCallback (internal/process/next_tick.js:139:11)
    at process._tickCallback (internal/process/next_tick.js:181:9)
```

## Reproducible Demo
1. install expo-cli using `npm install expo-cli -g`
2. start a new project using 'expo init`
3. cd into the project folder and run ```expo start``` or ```npm start```

**Comment by byCedric** [comment]
@KaizhaoLiang I've been able to replicate it on Node 8, but on the latest (Node 11) it just works. Might be something for you to try for your demo...

**Comment by smcguffi** [comment]
> @KaizhaoLiang I've been able to replicate it on Node 8, but on the latest (Node 11) it just works. Might be something for you to try for your demo...

I'm on v11.1.0 and still having this issue.

**Comment by tarikneradin** [comment]
I've tried v8.11.3 and v11.2.0, having the same issue.

**Comment by meatwallace** [comment]
> @KaizhaoLiang I've been able to replicate it on Node 8, but on the latest (Node 11) it just works. Might be something for you to try for your demo...

running 11.0 locally works, but 11.0 is failing in our CI

**Comment by allison-buhler** [comment]
I'm seeing this issue locally on my Macbook Pro with Node version 8.9.4, no CI integration to compare against and haven't tried Node v11 yet.

**Comment by sujithvellat** [comment]
I was setting up expo on my new laptop and I got the same error on a new project with node v11.2.0.
Here is the diagnostics report

Sujiths-MBP:personal sujithvellat$ expo init
? Choose a project name: MyApp
? Choose a template: blank
[17:47:48] Downloading project files...
[17:47:52] Extracting project files...
[17:47:58] Customizing project...

Your project is ready at /Users/sujithvellat/projects/personal/MyApp
To get started, you can type:

  cd MyApp
  expo start

Sujiths-MBP:personal sujithvellat$ cd MyApp/
Sujiths-MBP:MyApp sujithvellat$ expo start
[17:48:13] Starting project at /Users/sujithvellat/projects/personal/MyApp
[17:48:14] Expo DevTools is running at http://localhost:19004
[17:48:14] Opening DevTools in the browser... (press shift-d to disable)
**[17:48:15] TypeError: Cannot read property 'compile' of undefined**
Sujiths-MBP:MyApp sujithvellat$ expo diagnostics
[17:48:43] Generating diagnostics report...
[17:48:43] You can join our slack here: https://slack.expo.io/.

  Expo CLI 2.4.0 environment info:
    System:
      OS: macOS 10.14.1
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 11.2.0 - /usr/local/bin/node
      npm: 6.4.1 - /usr/local/bin/npm
    IDEs:
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^31.0.2 => 31.0.2 
      react: 16.5.0 => 16.5.0 
      react-native: https://github.com/expo/react-native/archive/sdk-31.0.0.tar.gz => 0.57.1 

  Diagnostics report:
    https://exp-xde-diagnostics.s3.amazonaws.com/marlabsinnovations-899a40d7-9f17-4512-8901-468fecc390b0.tar.gz

**Comment by byCedric** [comment]
I've opened a forum issue, sometimes they respond faster there. https://forums.expo.io/t/new-project-with-expo-cli-crashes/16310

**Comment by rsmelo92** [comment]
I'm facing the same issues when creating a new app with expo on 8 and also on 11

**Comment by zbaloch** [comment]
I am also facing the same issue. 

```
[02:18:25] Expo DevTools is running at http://localhost:19002
[02:18:25] Opening DevTools in the browser... (press shift-d to disable)
[02:18:30] TypeError: Cannot read property 'compile' of undefined
```

**Comment by bogdan-calapod** [comment]
I also tried installing webpack-dev-server for what it's worth (saw some older issues related to this) but didn't do anything.

**Comment by SunburntRock89** [comment]
Can repro, Node 10.13.0

**Comment by SunburntRock89** [comment]
Ran with DEBUG=true option, got the following output.
https://gist.github.com/SunburntRock89/250012020cd0cfbd273113688c28b0db

Tried to get diagnostic info but got ```D:\Documents\Projects\YT-App>expo diagnostics
[23:42:33] Generating diagnostics report...
[23:42:33] You can join our slack here: https://slack.expo.io/.

  Expo CLI 2.4.0 environment info:
    System:
      OS: Windows 10
    Binaries:
      npm: 6.4.1 - C:\Program Files\nodejs\npm.CMD

  Diagnostics report:
\[23:42:37] Request failed with status code 500
[23:42:37] Error: Request failed with status code 500
    at createError (C:\Users\rhys0\AppData\Roaming\npm\node_modules\expo-cli\node_modules\axios\lib\core\createError.js:16:15)
    at settle (C:\Users\rhys0\AppData\Roaming\npm\node_modules\expo-cli\node_modules\axios\lib\core\settle.js:17:12)
    at IncomingMessage.handleStreamEnd (C:\Users\rhys0\AppData\Roaming\npm\node_modules\expo-cli\node_modules\axios\lib\adapters\http.js:236:11)
    at IncomingMessage.emit (events.js:187:15)
    at endReadableNT (_stream_readable.js:1094:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

**Comment by albertstartup** [comment]
I tried expo-cli@2.0.0 on node 10.11, same problem.

**Comment by skyleraustin** [comment]
Same error - expo-cli 2.1.0 and 2.4.0 on both node 10.4.0 and 11.2.0

**Comment by marcosmartini** [comment]
Same here. 
expo-cli 2.3.8 and 2.4.0 on node 8.11.3 on Mac

**Comment by Sciscitator** [comment]
+1

node v10.5.0 and expo 2.4.0 on Macbook Pro

**Comment by Kweiza** [comment]
+1
I tried expo-cli 2.2.0, 2.2.4, 2.3.0 and 2.4.0 and node v8.12.0 on Ubuntu 18.04

**Comment by rsimpsonn** [comment]
+1
node v11.2.0, expo-cli 2.4.0 on Macbook Pro

**Comment by dachinat** [comment]
+1
<img width="510" alt="screen shot 2018-11-26 at 4 10 33 am" src="https://user-images.githubusercontent.com/22872749/48986660-464dde80-f131-11e8-85a7-8421126f598e.png">

**Comment by francisdelgad** [comment]
I'm also experiencing this bug

**Comment by guppythegod** [comment]
I literally came here to create this issue. I'm having the same problem and can't seem to fix it.

**Comment by shergea** [comment]
+1
node v10.13.0, expo-cli 2.4.0 on Windows 10
node v11.2.0, expo-cli 2.4.0 on Macbook Pro

**Comment by ScoRoc** [comment]
+1
had the same error with node v8.9.4.
upgraded to v11.2.0 and same error

npm v6.4.1 for both

Macbook Pro

**Comment by Cameron1996** [comment]
Got the same issue, did anyone else get any weird EOF issues before this? Code seemed to be working fine, then EOF issues and while trying to fix that im now no longer able to launch it at all.

**Comment by andresmanikis** [comment]
Same here, node 10.13.0, npm 6.4.1, expo-cli 2.4.0.

**Comment by lfernando2091** [comment]
I have the same error 
- TypeError: Cannot read property 'compile' of undefined

**Comment by Snakeyesz** [comment]
+1
Node: v9.11.1
npm: 6.1.0
Expo-cli: 2.4.0, 2.2.4

Was on 2.2.4, updated to 2.4.0, got this error, rolling back to 2.2.4 did not fix it.

**Comment by jMuzsik** [comment]
Someone screwed up

**Comment by johsire** [comment]
Same issues been trying to make this work since Friday!! Researching everywhere even installed webpack-dev-server as per an old post on stack-overflow but to no success! 

Code Editor: expo start command:
17:44:28] Expo DevTools is running at http://localhost:19002
[17:44:28] Opening DevTools in the browser... (press shift-d to disable)
[17:44:32] TypeError: Cannot read property 'compile' of undefined

Code Editor: npm start command:
[18:05:34] Expo DevTools is running at http://localhost:19002
[18:05:34] Opening DevTools in the browser... (press shift-d to disable)
[18:05:38] TypeError: Cannot read property 'compile' of undefined
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! empty-project-template@ start: `expo start`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the empty-project-template@ start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/johsire/.npm/_logs/2018-11-26T01_05_38_268Z-debug.log

Browser:
Expo Developer Tools is disconnected from Expo CLI. Use the expo start command to start the CLI again.

Using:  

- node v10.13.0; 
- npm v6.4.1; 
- expo-cli v2.4.0

**Comment by thewei** [comment]
+1
![image](https://user-images.githubusercontent.com/396570/48988063-3d273680-f15e-11e8-92db-30dfc7bb4694.png)

**Comment by AndresOchoaS** [comment]
Same problem here! 😒

**Comment by tuanbs** [comment]
I got same error 🙁

**Comment by jMuzsik** [comment]
@fson @sjchmiela @esamelson 

Apologise, you guys are likely not working right now, but this is obviously a major problem if it hasn't been noted as of yet.

**Comment by Sushobhan04** [comment]
Same problem. This error showed up after I installed expo-cli today. My friend who installed a week ago, says it works fine with his machine.

**Comment by SunburntRock89** [comment]
Yeah, my expo worked yesterday but when I reinstalled today it kept giving
me this error.

On 26 Nov 2018 01:49, "Sushobhan Ghosh" <notifications@github.com> wrote:

Same problem. This error showed up after I installed expo-cli today. My
friend who installed a week ago, says it works fine with his machine.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/expo/expo-cli/issues/213#issuecomment-441495854>, or mute
the thread

**Comment by apnoble** [comment]
Same issue running the latest version of node and expo cli.

**Comment by SunburntRock89** [comment]
Could someone try downgrading expo by 1 minor version?

**Comment by KIRPAT** [comment]
Same issue, I'm about to go crazy!

**Comment by apnoble** [comment]
I uninstalled expo-cli and then installed version 2.3.12. I created a new project with expo init and then still ran into the same issue.

**Comment by marcosmartini** [comment]
> Could someone try downgrading expo by 1 minor version?

I was running on 30 and it wasn't working.

**Comment by AndresOchoaS** [comment]
I have reviewed all dependencies at https://www.npmjs.com/package/expo-cli and none have been updated yesterday or today. This error is very strange.

**Comment by jMuzsik** [comment]
It actually worked for me when I installed it Friday. Then, I was having a different error tonight which led to me reinstalling it (the cli). This led to the current error. The only major thing I did with my computer is update my MacOS.

**Comment by andresmanikis** [comment]
I don’t think the problem is in one of expo-cli’s dependencies. I think it is in one of the dependencies of the project it generates.

**Comment by Gigiman** [comment]
Got same error while trying to run expo build:android.

**Comment by AndresOchoaS** [comment]
> I don’t think the problem is in one of expo-cli’s dependencies. I think it is in one of the dependencies of the project it generates.

Only expo, react, react-native and babel-preset-expo (dev) in an empty expo-cli project.

**Comment by jMuzsik** [comment]
This may be a server problem seeing as `expo diagnostics` returns a status code of 500

**Comment by jMuzsik** [comment]
So, rather then this being an actual problem with the code, it's actually a problem with the servers that expo cli utilises and we just have to wait until they start working again.

**Comment by ranggarifqi** [comment]
got the same error on node v8
still got the error after upgrading node to v11

npm v6.4.1
expo-cli v2.4.0

**Comment by sy92th** [comment]
Hi everyone, I have encountered this issue when upgrading expo-cli version to 2.4.0
Removing expo-cli and reinstall the old version does not solve this problem

Here's how I solved the problem:
- Uninstall the current version and install expo-cli version 2.2.5
- Delete folder expo-cli in /usr/local/lib/node_modules/ (Location of expo-cli when you install global module)
- Download expo-cli from this link (I copied from my friend's computer, he did not upgrade to the new version) and extract to the expo-cli folder.
`https://drive.google.com/file/d/1sIhS-yuLfFliwpds6xv1JaVQGb7PF0L1/view?usp=sharing`

**Comment by twise2** [comment]
+1

[22:26:18] Starting project at /Users/user/Desktop/app
[22:26:20] Expo DevTools is running at http://localhost:19002
[22:26:20] Opening DevTools in the browser... (press shift-d to disable)
[22:26:25] TypeError: Cannot read property 'compile' of undefined

MacOS Sierra, version 10.12
/usr/local/Cellar/watchman/4.9.0_2/bin/watchman
└── react-native@0.55.4
└── expo-cli@2.2.5

**Comment by muadem1992** [comment]
It worked for me @sy92th

**Comment by sislovesmedotcom** [comment]
Thank you @sy92th 👍

**Comment by il10** [comment]
What is the command to install a specific version for expo? 
In this case I am trying to install expo 2.2.5

**Comment by sy92th** [comment]
@il10 What is the command to install a specific version for expo?
`npm install -g expo-cli@2.2.5`

**Comment by dowayneforde** [comment]
Thanks, @sy92th

**Comment by AndrewSkinner16** [comment]
I've uninstalled 4.4 and installed 2.2.5 and still getting the same error...
There is a new version of expo-cli available (2.4.0).
You are currently using expo-cli 2.2.5
Run `npm install -g expo-cli` to get the latest version
[21:08:14] Starting project at /Users/andrewskinner/Desktop/test_react
[21:08:14] Expo DevTools is running at http://localhost:19002
[21:08:14] Opening DevTools in the browser... (press shift-d to disable)
[21:08:16] TypeError: Cannot read property 'compile' of undefined
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! my-new-project@ start: `expo start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the my-new-project@ start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/andrewskinner/.npm/_logs/2018-11-26T05_08_16_674Z-debug.log

**Comment by romnycristopher** [comment]
Worked for me, Thanks @sy92th !

**Comment by brightidea** [comment]
still having the same issue even after trying the fixes mentioned above. Had hope for @sy92th's to work but alas no dice.

**Comment by udnisap** [comment]
Full stack trace if anyone wants to dig deaper

```
VM558 log.js:56 [12:13:29] Starting project at /private/tmp/React-Native/AwesomeProject
VM558 log.js:56 [12:13:29] Expo DevTools is running at http://localhost:19002
VM558 log.js:56 [12:13:29] Opening DevTools in the browser... (press shift-d to disable)
VM558 log.js:84 [12:13:35] TypeError: Cannot read property 'compile' of undefined
consoleError @ VM558 log.js:84
(anonymous) @ VM558 log.js:166
respectProgressBars @ VM558 log.js:94
error @ VM558 log.js:165
_callee$ @ VM85 exp.js:284
tryCatch @ VM160 runtime.js:62
invoke @ VM160 runtime.js:296
prototype.(anonymous function) @ VM160 runtime.js:114
step @ VM86 asyncToGenerator.js:17
(anonymous) @ VM86 asyncToGenerator.js:30
Promise rejected (async)
step @ VM86 asyncToGenerator.js:27
(anonymous) @ VM86 asyncToGenerator.js:28
Promise resolved (async)
step @ VM86 asyncToGenerator.js:27
(anonymous) @ VM86 asyncToGenerator.js:35
F @ VM96 _export.js:36
(anonymous) @ VM86 asyncToGenerator.js:14
(anonymous) @ VM85 exp.js:306
listener @ VM161 index.js:315
emit @ VM21 events.js:180
Command.parseArgs @ VM161 index.js:654
Command.parse @ VM161 index.js:474
runAsync @ VM85 exp.js:654
_callee5$ @ VM85 exp.js:723
tryCatch @ VM160 runtime.js:62
invoke @ VM160 runtime.js:296
prototype.(anonymous function) @ VM160 runtime.js:114
step @ VM86 asyncToGenerator.js:17
(anonymous) @ VM86 asyncToGenerator.js:35
F @ VM96 _export.js:36
(anonymous) @ VM86 asyncToGenerator.js:14
run @ VM85 exp.js:731
(anonymous) @ VM82 expo.js:11
Module._compile @ VM43 loader.js:651
Module._extensions..js @ VM43 loader.js:665
Module.load @ VM43 loader.js:566
tryModuleLoad @ VM43 loader.js:506
Module._load @ VM43 loader.js:498
Module.runMain @ VM43 loader.js:695
startup @ VM20 node.js:201
bootstrapNodeJSCore @ VM20 node.js:516
```

**Comment by ankraj077** [comment]
Same here

`TypeError: Cannot read property 'compile' of undefined`

This is so frustrating

Wasting hours of time to solve this types of errors instead of doing some real work on app features.

This is completely demotivating me to work on my stuff. Expo has bunch of such issues. Not good.

**Comment by kot-ezhva** [comment]
2.2.5 is't worked for me.
I tried:
Reinstall yarn
Reinstall expo (yarn global add expo-cli@2.2.5)
Result via `expo start`: 
<img width="543" alt="image" src="https://user-images.githubusercontent.com/15121885/49001711-97c88e80-f16e-11e8-8940-1e8081cd7a6b.png">
😿

**Comment by bonham000** [comment]
Same issue for my, described here (https://forums.expo.io/t/running-start-and-publish-with-expo-cli-throws-typeerror/16310/6).

I get this issue now repeatedly in CI builds, but I can run expo locally without issue. Expo cli version 2.4.0 in both environments. CI is running node 8.11.3 in docker.

A co-worker is having the same failure locally when running `expo start`.

**Comment by sjchmiela** [comment]
Hey all, thank you for your patience and all the proposed workarounds, solutions and explanations! We really appreciate that the community is working hand in hand to help each other while we're asleep. 😅 

I was able to reproduce the problem, I'll see what I can do to fix it ASAP! 🙂

**Comment by pawel-ogorzalek** [comment]
Hi, same issue for me. I posted [StackOverflow issue](https://stackoverflow.com/questions/53472707/cannot-read-property-compile-of-undefined-failed-to-start-new-react-native-pr) yesterday.

**Comment by blucell** [comment]
It seems to be a problem with an update of a dependency, thus reinstalling it causes the problem, because it upgrades its minor versions. After inspection:

`node --inspect ~/.config/yarn/global/node_modules/expo-cli/bin/expo.js start`

![screen shot 2018-11-26 at 11 08 41](https://user-images.githubusercontent.com/4368958/49003910-af524800-f16b-11e8-98bb-baa86c5b3032.png)

Which is in the file `~/.config/yarn/global/node_modules/xdl/node_modules/joi/lib/index.js`

**Comment by sjchmiela** [comment]
That's right! We've just found this too! `joi` was upgraded 14 hours ago with version 14.3.0 which produces this issue, we'll publish `expo-cli` with `joi` dependency requirement fixed soon. 🙂

**Comment by MuhammetAliDere** [comment]
+1
os: mac high sierra 10.13.4
node: 11.2.0
npm: 6.4.1
expo: 2.4.0
<img width="557" alt="screen shot 2018-11-26 at 12 16 11" src="https://user-images.githubusercontent.com/10808460/49004354-29d39580-f175-11e8-92ab-50062b6ed605.png">

**Comment by incNick** [comment]
> `expo-cli@2.4.1` published! A fresh install of `expo-cli`, i. e.
> 
> ```shell
> npm uninstall -g expo-cli && npm i -g expo-cli
> ```
> 
> solves the issue (I don't know if plain `npm i -g expo-cli` works too, but it should).

Great~

**Comment by TheSalarKhan** [comment]
> `expo-cli@2.4.1` published! A fresh install of `expo-cli`, i. e.
> 
> ```shell
> npm uninstall -g expo-cli && npm i -g expo-cli
> ```
> 
> solves the issue (I don't know if plain `npm i -g expo-cli` works too, but it should). slightly_smiling_face

Thanks @sjchmiela. I confirm that this now works like a dream. :slightly_smiling_face: :tada: 

Closing the issue.

**Comment by byCedric** [comment]
@latifuluman yes, the issue was related to `joi@14.3.0` and `expo-cli@2.4.1` (as stated above). After the patch was released in `expo-cli@2.4.1`, a "clean install" of the CLI was recommended to patch all expo projects and solve the issue.

If you encounter this problem again in all later CLI versions, let us know! 😀 And if so, could you share the version of your `expo-cli` in a new issue? For now, happy coding! 😁
